modules/common/src/encoding.ts

```diff
@@ -102,6 +102,35 @@ export function decodeRatio(
   return Number(onChainValue) / (10 ** digits);
 }
 
+/** Encodes the particles of a proposalId into a globally unque value for subgraph */
+export function encodeProposalId(pluginAddress: string, id: number) {
+  if (!/^0x[A-Fa-f0-9]{40}$/.test(pluginAddress)) {
+    throw new Error("Invalid address");
+  }
+
+  return `${pluginAddress}_0x${id.toString(16)}`;
+}
+
+/** Decodes a proposalId from subgraph and returns the original pluginAddress and the nonce */
+export function decodeProposalId(
+  proposalId: string,
+): { pluginAddress: string; id: number } {
+  if (!/^0x[A-Fa-f0-9]{40}_(0x[A-Fa-f0-9]{1,64})$/.test(proposalId)) {
+    throw new Error("Invalid proposalId");
+  }
```

Comment:
That doesn't follow the other functions that are implemented in `@aragon/sdk-common`. The common package defines custom errors but all it's functions throws "normal" errors with a message.
For example in `modules/common/src/encoding.ts` lines 1 - 10:
```typescript
/** Decodes a hex string and returns it as a buffer */
export function hexToBytes(hexString: string): Uint8Array {
  if (!hexString) return new Uint8Array();
  else if (!/^(0x)?[0-9a-fA-F]*$/.test(hexString)) {
    throw new Error("Invalid hex string");
  } else if (hexString.length % 2 !== 0) {
    throw new Error("The hex string has an odd length");
  }
```

Reply:
That's a good point, we have a low priority task for refactoring these generic errors into specific ones. 

At the very least, we should be calling `isProposalId()` instead of copying the regex every time

Reply:
`isProposalId()` is part of the common client in client and not part of the common package.

Reply:
Then I would move the function and any related testing here instead. 

There should be only one source of truth, regarding valid/invalid proposal ID's, and the existing logic shouldn't be impacting any future plugins using hashing instead of nonce